File: reporter_backend/app/schemas/tickets.py
from typing import List, Optional, Any
from pydantic import BaseModel

class TicketRow(BaseModel):
    id_venta: int
    CNOMBREAGENTE: Optional[str]
    serie: Optional[str]
    folio: int
    fecha: str  # Fecha formateada o raw
    hora: str
    subtotal: float
    descuento: float
    impuesto: float
    total: float
    fEfectivo: float
    Tarjeta: float
    fVales: float
    fTrans: float
    fOtro: float
    fechaCancelado: Optional[str]
    cancelado: str
    CNOMBREALMACEN: str

class TicketKpis(BaseModel):
    total_vendido: float
    total_tickets: int
    total_cancelados: int
    promedio_ticket: float

class VentasPorAgenteItem(BaseModel):
    agente: str
    total: float
    cantidad_tickets: int

class TicketsResponse(BaseModel):
    data: List[TicketRow]
    total: int
    page: int
    size: int
    kpis: TicketKpis

File: reporter_frontend/src/types/tickets.ts
export interface TicketRow {
    id_venta: number;
    CNOMBREAGENTE: string;
    serie: string | null;
    folio: number;
    fecha: string;
    hora: string;
    subtotal: number;
    descuento: number;
    impuesto: number;
    total: number;
    fEfectivo: number;
    Tarjeta: number;
    fVales: number;
    fTrans: number;
    fOtro: number;
    fechaCancelado: string | null;
    cancelado: 'cobrado' | 'cancelado';
    CNOMBREALMACEN: string;
}

export interface TicketKpis {
    total_vendido: number;
    total_tickets: number;
    total_cancelados: number;
    promedio_ticket: number;
}

export interface TicketsResponse {
    data: TicketRow[];
    total: number;
    page: number;
    size: number;
    kpis: TicketKpis;
}

export interface VentasPorAgenteItem {
    agente: string;
    total: number;
    cantidad_tickets: number;
}
